capstone/src/main/webapp/youtubeScript.js

```diff
@@ -0,0 +1,176 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const ATTRIBUTES_BY_LANGUAGE = {
+  'en': ['TOXICITY', 'SEVERE_TOXICITY', 'TOXICITY_FAST', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT', 'SEXUALLY_EXPLICIT', 'FLIRTATION'],
+  'es': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'fr': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'de': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'it': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
+};
+
+/** Calls youtube servlet and passes output to perspctive */
+async function callYoutube() {
+  const channelId = document.getElementById('channelIdForAnalysis').value.replace(/ /g, '');
+  if (!channelId) {
+    return;
+  }
+  var response;
+  /** Checks if input follows channel ID format, if not attempts to convert it to channel ID*/
+  if (channelId[0] == "U" && channelId[1] == "C" && channelId.length == 24 && isLetter(channelId[channelId.length-1])) {
+    response = await fetch('/youtube_servlet?channelId=' + channelId,)
+    response = await response.json();
+    if (response.hasOwnProperty('error')) {
+      alert("Invalid Channel ID");
+      inputCommentsToPerspective([]);
+      return;
+    }
+  } else {
+    const usernameConverterResponse = await fetch('/username_servlet?channelId=' + channelId,)
+    const usernameConverterResponseJson = await usernameConverterResponse.json();
+    if (usernameConverterResponseJson.pageInfo.totalResults == 0) {
+      alert("Username Not found, Please Input Channel ID");
+      inputCommentsToPerspective([]);
+      return;
+    }
+    const convertedUserName = usernameConverterResponseJson.items[0].id;
+    response = await fetch('/youtube_servlet?channelId=' + convertedUserName,)
+    response = await response.json();
+  }
+  inputCommentsToPerspective(response);
+}
+
+/** Calls perspective to analyze comments */
+async function inputCommentsToPerspective(comments) {
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+  const commentListElement = document.getElementById('comment-list');
+  commentListElement.innerHTML = '';
+  const requestedAttributes = getRequestedAttributes();
+  if (!requestedAttributes) {
+      return;
+  }
+  const attributeScores = [];
+  for (const item in comments.items) {
+    const perspectiveScore = await callPerspective(comments.items[item].snippet.topLevelComment.snippet.textOriginal, langElement.value, requestedAttributes);
```

Comment:
(I made this same comment in #15 - maybe we can talk about this at one of our pod syncs)

This isn't a big deal for now, but I believe the way await works in this case is that each callPerspective needs to be completed before the next one runs, which might slow down the UX a bit. There are other ways you can write this, including using `Promise` or `Promise.all()`, where you can make all perspective requests in parallel, which may speed things up (you might have to be careful about variables used in the loop). This page has a little more info: https://stackoverflow.com/questions/11488014/asynchronous-process-inside-a-javascript-for-loop

Maybe for now you can add a `TODO` comment saying something like these requests could be made in parallel if we need to speed up the webpage

Reply:
I have incorporated the use of a promise to enable perspective requests in parallel.
@chrismohrii your code was a great reference.